### PR TITLE
Add option to select sync or async query execution in the backend.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

**Review by havardpe — COMMENTED** [review]
looks good, please consider my comments

File: searchcore/src/vespa/searchcore/config/proton.def [hunk 1]
```diff
@@ -22,6 +22,10 @@ distributionkey int default=-1
 ## Number of threads used for rpc transport threads
 rpctransportthreads int default=4 restart
 
+## Dispatch requests to threadpool
+## If false, rpctransportthreads will be set to num cores
+asyncsearch bool default=true
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/config/proton.def]: Fixed

File: searchcore/src/vespa/searchcore/config/proton.def [hunk 2]
```diff
@@ -22,6 +22,10 @@ distributionkey int default=-1
 ## Number of threads used for rpc transport threads
 rpctransportthreads int default=4 restart
 
+## Dispatch requests to threadpool
+## If false, rpctransportthreads will be set to num cores
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/config/proton.def]: Fixed

**Comment by baldersheim** [comment]
@havardpe Now comments are addressed PTAL.

**Review by havardpe — APPROVED** [review]

**Pull request merged**